pkg/agent/agent.go

```diff
@@ -695,9 +703,7 @@ func (i *Initializer) configureGatewayInterface(gatewayIface *interfacestore.Int
 		klog.Errorf("Failed to find host link for gateway %s: %v", i.hostGateway, err)
 		return err
 	}
-
-	i.nodeConfig.GatewayConfig = &config.GatewayConfig{Name: i.hostGateway, MAC: gwMAC, OFPort: uint32(gatewayIface.OFPort)}
-	gatewayIface.MAC = gwMAC
+	i.configureGatewayMAC(gatewayIface, gwMAC)
```

Comment:
The function also assigns gatewayConfig to `i.nodeConfig` which cannot be told from the function name.

Besides, I wonder if it could happen on linux that the ovs db doesn't have mac_in_use set when antrea-agent reads it if the field is loaded asynchonously. And to make newly created antrea-gw0 and existing antrea-gw0 consistent, I feel we could unify the code across OSes as following:
1. Read "mac" from ovsdb, instead of "mac_in_use"
2. Always assign "mac" from gatewayIface to "i.nodeConfig.GatewayConfig" regardless of platforms
3. For existing antrea-gw0, "mac" in gatewayIface would be empty, use mac got from "setLinkUp" to update gatewayIface and update ovsdb to reflect it.

Reply:
On Windows, the `MAC` field configured in Interface is not the actual value used in network adapter. So we can not use OVSDB as the single source for both Linux and Windows.